app/models/deleted_resource.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+##
+# Model to store deleted resources.
+# These may be used in order to serve '410 Gone' statuses.
+class DeletedResource
+  include Mongoid::Document
+
+  belongs_to :deleted_by, class_name: 'User', optional: true, inverse_of: :deleted_resources,
+                          index: true
+
+  field :resource_type, type: String
+  field :resource_uuid, type: String
+  field :deleted_at, type: Time
```

Comment:
Could the type be changed to `DateTime` to be consistent with `Contribution#first_published_at`?